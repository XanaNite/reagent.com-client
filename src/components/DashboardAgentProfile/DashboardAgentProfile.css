.dashboardAgent-agentInfo{
    margin-bottom: 100px;
}

.dashboardAgent-profile{
    display: flex;
}

.dashboardAgent-profile-img{
    font-size: 100px;
}

.dashboardAgent-name{
    height: 100px;
    padding-left: 15px;
}

.dashboardAgent-name h2{
    margin-top: 10px;
    margin-bottom: 10px;
    font-weight: 500;
    font-size: 18px;
}

.dashboardAgent-name p{
    margin-top: 10px;
    margin-bottom: 10px;
}

.dashboardAgent-contact table{
    margin-top: 25px;
    width: 100%;
}

.tableStyle{
    font-weight: 500;
}