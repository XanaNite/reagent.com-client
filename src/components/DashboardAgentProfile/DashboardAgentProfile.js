import React from 'react'
import {Link} from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUser } from '@fortawesome/free-solid-svg-icons'
import './DashboardAgentProfile.css'

export default function DashboardAgentProfile(props){
    return(
        <div className="dashboardMain-container">
            <div className="dashboardAgent-agentInfo">
                <div className="dashboardAgent-profile">
                    <div className='dashboardAgent-profile-img'>
                        <FontAwesomeIcon icon={faUser} />
                    </div>
                    <div className="dashboardAgent-name">
                        <h2>{props.agent.first_name} {props.agent.last_name}</h2>
                        <p>"{props.agent.slogan}"</p>
                    </div>
                </div>
                <div className="dashboardAgent-contact">
                    <table>
                        <tbody>
                            <tr>
                                <td className="tableStyle">Email</td>
                                <td>{props.agent.agent_email}</td>
                            </tr>
                            <tr>
                                <td className="tableStyle">Phone</td>
                                <td>{props.agent.agent_phone_type} {props.agent.agent_phone}</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
            <Link to='/myprofile'>Improve your profile</Link>
        </div>
    )
}