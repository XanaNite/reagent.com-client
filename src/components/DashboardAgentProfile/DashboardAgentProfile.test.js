import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DashboardAgentProfile from './DashboardAgentProfile';

describe(`DashboardAgentProfile component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the DashboardAgentProfile given props', () => {
        const wrapper = shallow(<DashboardAgentProfile {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})