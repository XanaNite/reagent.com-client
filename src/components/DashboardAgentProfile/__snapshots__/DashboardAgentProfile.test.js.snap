// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardAgentProfile component renders the DashboardAgentProfile given props 1`] = `
<div
  className="dashboardMain-container"
>
  <div
    className="dashboardAgent-agentInfo"
  >
    <div
      className="dashboardAgent-profile"
    >
      <div
        className="dashboardAgent-profile-img"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f007",
                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
              ],
              "iconName": "user",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <div
        className="dashboardAgent-name"
      >
        <h2>
          test
           
          test
        </h2>
        <p>
          "
          "
        </p>
      </div>
    </div>
    <div
      className="dashboardAgent-contact"
    >
      <table>
        <tbody>
          <tr>
            <td
              className="tableStyle"
            >
              Email
            </td>
            <td />
          </tr>
          <tr>
            <td
              className="tableStyle"
            >
              Phone
            </td>
            <td>
               
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <Link
    to="/myprofile"
  >
    Improve your profile
  </Link>
</div>
`;
